jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MojoCommon.java

```diff
@@ -53,18 +53,20 @@
   public static final String VERSION_URL = "https://storage.googleapis.com/jib-versions/jib-maven";
 
   static Future<Optional<String>> newUpdateChecker(
-      ProjectProperties projectProperties, Log logger) {
-    if (projectProperties.isOffline() || !logger.isInfoEnabled()) {
+      ProjectProperties projectProperties, GlobalConfig globalConfig, Log logger) {
```

Comment:
maybe we can pass in a boolean `isDisableUpdateCheck` instead of a GlobalConfig as a parameter? It is only used once to check `globalConfig.isDisableUpdateCheck()`.

Reply:
Technically you're right. And the same argument would apply to `logger` (it's not used elsewhere either). But as you see, the method signature accepts `logger`, not `isInfoLogEnabled`. And `projectProperties` is also aligned with this. So this is more about an API design. As `MojoCommon` is a helper class that just shares duplicate code between `~Mojo` classes, I think this pattern is reasonable.